src/pages/WorkloadDetails/WorkloadInfo/WorkloadPodLogs.tsx

```diff
@@ -0,0 +1,238 @@
+            position: 'absolute',
+            zIndex: 9999
+          }}
+        >
+          <div id="wpl_header" className="modal-header">
+            <Button className="close" bsClass="" onClick={this.props.onClose}>
+              <Icon title="Close" type="pf" name="close" />
+            </Button>
+            <span className="modal-title">Pod Logs: {this.props.podName}</span>
+          </div>
+          {this.state.container && (
+            <>
+              <Toolbar>
+                <ToolbarDropdown
+                  id={'wpl_containers'}
+                  nameDropdown="Container"
+                  tooltip="Display logs for the selected pod container"
+                  handleSelect={key => this.setContainer(key)}
+                  value={this.state.container}
+                  label={this.state.container}
+                  options={this.state.containers!}
+                />
+                <Toolbar.RightContent>
+                  <ToolbarDropdown
+                    id={'wpl_duration'}
+                    handleSelect={key => this.setDuration(key)}
+                    value={this.state.duration}
+                    label={Durations[this.state.duration]}
+                    options={Durations}
+                    tooltip={'Time range for graph data'}
+                  />
+                  <span style={{ paddingLeft: '0.5em' }}>
+                    <Button id={'wpl_refresh'} disabled={!this.state.podLogs} onClick={() => this.handleRefresh()}>
+                      <Icon name="refresh" />
+                    </Button>
+                  </span>
+                </Toolbar.RightContent>
+              </Toolbar>
+              <textarea
+                style={{
+                  width: '100%',
+                  height: '100%',
+                  resize: 'vertical',
+                  color: '#fff',
+                  backgroundColor: '#003145'
+                }}
+                readOnly={true}
+                value={this.state.podLogs ? this.state.podLogs.logs : 'Loading logs...'}
+              />
+            </>
+          )}
+          {this.state.loadingPodError && <div>{this.state.loadingPodError}</div>}
+        </div>
+      </Draggable>
+    );
+  }
+
+  private setContainer = (container: string) => {
+    this.setState({ container: container });
+  };
+
+  private setDuration = (duration: string) => {
+    this.setState({ duration: duration });
+  };
+
+  private handleRefresh = () => {
+    this.fetchLogs(this.props.namespace, this.props.podName, this.state.container!, Number(this.state.duration));
+  };
+
+  private fetchPod = (namespace: string, podName: string) => {
+    const promise: Promise<Response<Pod>> = getPod(namespace, podName);
+    this.loadPodPromise = makeCancelablePromise(Promise.all([promise]));
+    this.loadPodPromise.promise
+      .then(response => {
+        const pod = response[0].data;
+        const containers: Object = {};
+        if (pod.containers) {
```

Comment:
i think code could be a little bit more concise if `pod.containers` and `pod.istioContainers` were not optional (empty arrays instead) - which means to initialize them explicitly from backend rather than relying on golang's default value for slices.

Reply:
I started making this change but the implications to existing UI code became a bit overwhelming as there is quite a lot of code already expecting that these arrays may be unset.  So I'll suffer with the two additional guards here...